src/components/Destinations/DestinationConfigInput.tsx

```diff
@@ -0,0 +1,283 @@
+
+export interface DestinationConfigInputprops {
+  specs: Array<any>;
+  registerFormFieldValue: (...args: any) => any;
+  control: any;
+  setFormValue: (...args: any) => any;
+  unregisterFormField: (...args: any) => any;
+}
+
+export const DestinationConfigInput = ({
+  specs,
+  registerFormFieldValue,
+  control,
+  setFormValue,
+  unregisterFormField,
+}: DestinationConfigInputprops) => {
+  const [connectorSpecs, setConnectorSpecs] = useState<any>([]);
+  const [showPasswords, setShowPasswords] = useState<any>({});
+
+  const handleClickShowPassword = (field: string) => {
+    const tempShowPasswords: any = { ...showPasswords };
+    tempShowPasswords[field] = !showPasswords[field];
+    setShowPasswords(tempShowPasswords);
+  };
+
+  const handleObjectFieldOnChange = (
+    dropDownVal: any,
+    field: string,
+    fieldOnChangeFunc: any
+  ) => {
+    fieldOnChangeFunc.onChange(dropDownVal);
+
+    // Fetch the current selected spec of type object based on selection
+    const selectedSpec: any = connectorSpecs.find(
+      (ele: any) => ele.field === field
+    );
+
+    // Filter all specs that are under selectedSpec and have parent as selectedSpec
+    // Check if any child specs has type object
+    const filteredChildSpecs: any = [];
+    selectedSpec?.specs.forEach((ele: any) => {
+      if (ele.parent === dropDownVal) {
+        // Check if the child has another level or not
+        if (ele?.specs && ele?.enum.length === 0) {
+          ele.specs.forEach((childEle: any) => {
+            filteredChildSpecs.push({ ...childEle, order: ele.order });
+          });
+        } else {
+          filteredChildSpecs.push(ele);
+        }
+      }
+    });
+
+    // Set the order of child specs to be displayed at correct position
+    filteredChildSpecs.forEach((ele: any) => {
+      ele.order = selectedSpec.order;
+    });
+    // Update the specs state
+
+    // Find the specs that will have parent in the following enum array
+    const enumsToRemove: string[] = selectedSpec?.enum.filter(
+      (ele: any) => ele !== dropDownVal
+    );
+
+    const tempSpecs = connectorSpecs
+      .filter((sp: any) => !enumsToRemove.includes(sp?.parent))
+      .concat(filteredChildSpecs);
+
+    // Unregister the form fields that have parent in enumsToRemove
+    connectorSpecs.forEach((sp: any) => {
+      if (enumsToRemove.includes(sp?.parent)) unregisterFormField(sp?.field);
+    });
+
+    setConnectorSpecs(tempSpecs);
+  };
+
+  useEffect(() => {
+    setConnectorSpecs(specs);
+  }, [specs]);
+
+  useEffect(() => {
+    const tempShowPasswords: any = {};
+    specs.forEach((element) => {
+      if (element?.airbyte_secret) {
+        tempShowPasswords[element.field] = false;
+      }
+    });
+    setShowPasswords(tempShowPasswords);
+  }, [specs]);
+  return (
+    <>
+      {connectorSpecs
+        .sort((input1: any, input2: any) => {
+          if (!input2.order) return -1;
```

Comment:
Not sure why this is used?

Reply:
Airbyte has a `order` key to define which fields to show on top , basically define the sequence in which input fields should be shown. But in postgres one of the fields doesn't have this `order` so i want to put it in the end but while sorting it comes up on the top. hence i am doing to define sorting for when the `order` is not defined.